src/main.rs

```diff
@@ -315,6 +338,62 @@ impl BloomFilter {
     }
 }
 
+
+
+fn compute_bloom_size(fp_rate: f64, expected_ngram_count: usize) -> usize {
+    /* Uses binary search to find optimal size of bloom filter using optimal number of hashers
+       and provided ngram counts
+    */
+    // compute 90% of system ram 
+    let mut sys = System::new_all();
+    sys.refresh_all();
+
+
+    let mut lo = 1 as usize;
+    let mut hi = ((sys.total_memory() as f64) * 0.9) as usize;
+
+    // Save some time by checking endpoint first
+    if BloomFilter::prob_of_false_positive(hi, expected_ngram_count, 
+                                           BloomFilter::optimal_number_of_hashers(hi, expected_ngram_count)) > fp_rate {
+        return hi;
+    }
+
+    // Then do binary search to find optimal size
+    while lo < hi-1 { // -1 here because binsearch powers of 2 scare me 
+        let mid = lo + (hi - lo) / 2;
+        let num_hashers = BloomFilter::optimal_number_of_hashers(mid, expected_ngram_count);
+        let computed_fp = BloomFilter::prob_of_false_positive(mid, expected_ngram_count, num_hashers) ;
+        if computed_fp > fp_rate {
+            // FP rate too high, need to go bigger
+            lo =  mid + 1;
+        } else {
+            // FP rate too low, can make bloom filter smaller
+            hi = mid -1;
+        }
+    }
+    hi
+}
+
+
+fn expand_dirs(paths: &[PathBuf]) -> Result<Vec<PathBuf>> {
+    let mut files = vec![];
+    for path in paths {
+        if path.is_dir() {
+            let path_str = path
+                .to_str()
+                .ok_or_else(|| anyhow!("invalid path '{}'", path.to_string_lossy()))?;
+            for entry in glob(&format!("{}/**/*.json*.gz", path_str))? {
+                files.push(entry?.to_path_buf());
+            }
```

Comment:
I'm fine with this, but there is probably a more stream-like way of doing this.

Reply:
I... stole this code from @epwalsh directly (from a hotfix to wimbd)